src/main/java/com/puppycrawl/tools/checkstyle/checks/design/HideUtilityClassConstructorCheck.java

```diff
@@ -248,10 +248,7 @@ public boolean isHasPublicCtor() {
          */
         public void invoke() {
             final DetailAST objBlock = ast.findFirstToken(TokenTypes.OBJBLOCK);
-            hasNonStaticMethodOrField = false;
-            hasNonPrivateStaticMethodOrField = false;
             hasDefaultCtor = true;
-            hasPublicCtor = false;
```

Comment:
This method is the first thing called after the class is instantiated, so this might as well be part of the constructor. All these assignments are the already default values assigned in the constructor, so there is no reason to re-assign them.